src/update-repo-task-status.ts

```diff
@@ -0,0 +1,53 @@
+import { getState } from "@actions/core";
+
+import {
+  setVariantAnalysisCanceled,
+  setVariantAnalysisFailed,
+} from "./gh-api-client";
+import {
+  getControllerRepoId,
+  getRepos,
+  getVariantAnalysisId,
+  getWorkflowStatus,
+} from "./inputs";
+
+/**
+ * If the overall variant analysis workflow failed or was canceled,
+ * propagate the failure/cancellation status to the individual repo tasks.
+ */
+async function updateRepoTaskStatus(): Promise<void> {
+  const controllerRepoId = getControllerRepoId();
+  const variantAnalysisId = getVariantAnalysisId();
+  const workflowState = getWorkflowStatus();
+
+  if (!controllerRepoId || !variantAnalysisId || !workflowState) {
+    return;
+  }
+
+  const repos = getRepos();
+
+  for (const repo of repos) {
+    if (getState(`repo_${repo.id}_completed`)) {
+      continue;
+    }
```

Comment:
Note to self:

It looks like the `saveState`/`getState` functions only work for post-actions, so they don't work here: 
![image](https://user-images.githubusercontent.com/42641846/198687857-7c3b040e-ae7b-4721-8e3b-74b250efaa7f.png)

Perhaps we could call the "get repo task" API here to check whether a repo task has been completed already.

Reply:
(or we can pass the status along as outputs - see @robertbrignull's comment in the internal issue)

Reply:
Ooh, using the "get repo task" API is a good idea though. That way we'd be acting on more canonical data, rather than just what state changes we think we made already in the workflow. It would reduce the chance of weird edge-case bugs.

The only downside is it would increase the number of API request we make when a workflow is cancelled, but not by that much. I think it would add one request per repo, so we'd be doing 1-2 requests per repo instead of 0-1 requests. But I think we can handle this and it's not a problem. If we see scalability issues in the future we'll look at improving things, but right now I don't see any huge problems.

So right now I think we could go with either approach. Whatever you fancy implementing.